apps/citizen_advocate/web/models/user.ex

```diff
@@ -59,10 +63,46 @@ defmodule CitizenAdvocate.User do
   defp put_pass_hash(changeset) do
     case changeset do
       %Ecto.Changeset{valid?: true, changes: %{password: password}} ->
-        put_change(changeset, :password_hash, Comeonin.Bcrypt.hashpwsalt(password))
+        put_change(changeset, :password_hash, hashpwsalt(password))
       _ ->
         changeset
     end
   end
 
+  defp downcase_email(changeset) do
+    email = get_change(changeset, :email)
+    if is_nil(email) do
+      changeset
+    else
+      put_change(changeset, :email, String.downcase(email))
+    end
+  end
+
+  def confirmation_changeset(user, params = %{"token" => token}) do
+    user
+    |> cast(params, [])
+    |> put_change(:hashed_confirmation_token, nil)
+    |> put_change(:confirmed_at, Ecto.DateTime.utc())
+    |> validate_token(token)
+  end
+
+  defp validate_token(changeset, token) do
+    if checkpw(token, changeset.data.hashed_confirmation_token) do
+      changeset
+    else
+      add_error(changeset, :confirmation_token, "invalid")
+    end
+  end
+
+  def confirmation_needed_changeset(changeset) do
+    token = SecureRandom.urlsafe_base64(64)
```

Comment:
No need for the `SecureRandom` dependency, it's really simple with vanilla erlang/elixir: 

```elixir
:crypto.strong_rand_bytes(64) |> Base.encode64(case: :lower, padding: true)
```